### ISSUE TITLE
Review and revise saving to a repo you don't have writing permissions to.

### ISSUE THREAD

**Issue by ilovan** [cwrc/CWRC-GitWriter#51]
## Expected Behaviour

Check for writing permissions before going through (this repo exists, etc.)

## Steps to Reproduce (for bugs)

1. Open a repo you don't have writing permission to (e.g. https://dev-cwrc-writer.cwrc.ca/?githubPath=document.xml&githubRepo=SusanBrown%2FDTOC-Template)
2. Make a change
3. Hit Save on the ribbon
Result:

**Comment by ajmacdonald** [comment]
As part of this rework, I'm wondering if it makes sense to keep the additional "this repository exists - would you like to use it" confirmation? @ilovan

**Comment by ilovan** [comment]
can you condense it into "this repository and document path exist - would you like to use them?

**Comment by ajmacdonald** [comment]
Not easily because those are two separate checks that happen at different points in the saving process. Not to mention the case where the repository exists but the document path doesn't.

I just think it seems a bit redundant to ask the user if they really want to use the repository that they've already specified.

**Comment by ilovan** [comment]
right, but at the same time, the users should be warned if they are creating a new repo. So maybe instead of "this repository exists - would you like to use it" how about if the message "the repository you specified does not exist yet - would you like to create it?" (basically reverse the check)
maybe do the same for the second check as well.

**Comment by ajmacdonald** [comment]
I agree.

**Comment by ilovan** [comment]
actually, the second file already follows the logic I just described for the repo check - so no need to change that.

**Comment by ajmacdonald** [comment]
https://github.com/cwrc/cwrc-git-dialogs/commit/ec18e26c09a21797a5f599d97353e4728f13649e

**Comment by ilovan** [comment]
confirmed on dev-cwrc-writer
